### ISSUE TITLE
SQLite Disk I/O Error

### ISSUE THREAD

**Issue by leonardofalk** [rhomobile/rhodes#1007]
`could not prepare statement: 10; Message: disk I/O error`

Sometimes we get this message after doing a count query.

Sample model that triggers the error:

```ruby
class Parametro
  include Rhom::FixedSchema

  property :id_parametro, :string
  property :valor,        :string
end
```

```ruby
Parametro.count # this triggers the error
```

This was introduced after upgrading from 6.1 (gem, not git) to master branch (commit 1225a60826c0768129802606ae370d1b324c1ae5), as we're testing x64 Android builds. After digging a while some people say this is because the database journal file (ending in -journal) is not writable. I will try to keep this updated if I find anything more useful.

**Comment by alex-epifanoff** [comment]
@leonardofalk what is the device you observe this issue on?
Also could you please point to discussion where journal file is pointed at as a reason.

**Comment by leonardofalk** [comment]
@alex-epifanoff Mostly Android devices, any version.

Discussions:

- https://stackoverflow.com/a/53271447
- http://sqlite.1065341.n5.nabble.com/bug-failure-to-write-journal-reported-as-quot-disk-I-O-error-quot-td97788.html
- https://forums.gfi.com/index.php?t=msg&th=26707&goto=110816&

**Comment by alex-epifanoff** [comment]
fixed by 2cbb916db610930b031be499ef7fe3d9ac9d817a
